src/main/java/com/bekzat/gym/controller/TrainerController.java

```diff
@@ -0,0 +1,69 @@
+package com.bekzat.gym.controller;
+
+import com.bekzat.gym.dto.*;
+import com.bekzat.gym.service.TrainerService;
+import lombok.AllArgsConstructor;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.format.annotation.DateTimeFormat;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.*;
+
+import java.time.LocalDateTime;
+import java.util.List;
+
+@RestController
+@RequestMapping("/trainers")
+@AllArgsConstructor
+public class TrainerController {
+    @Autowired
+    private TrainerService trainerService;
+
+    @GetMapping
+    public List<TrainerReadDto> getTrainers(@RequestParam(required = false) String unassignedTraineeUsername) {
+        if (unassignedTraineeUsername != null) {
+            return trainerService.getTrainersNotAssignedToTrainee(unassignedTraineeUsername);
+        }
+        return trainerService.findAll();
+    }
+    @PostMapping("/register")
+    public ResponseEntity<CredentialsDto> registerTrainee(@RequestBody TrainerRegistrationDto registrationDto) {
+        CredentialsDto credentials = trainerService.register(registrationDto);
+        return new ResponseEntity<>(credentials, HttpStatus.CREATED);
+    }
+
+    @GetMapping("/{username}")
+    public ResponseEntity<TrainerProfileReadDto> getTrainerProfile(@PathVariable String username) {
+        TrainerProfileReadDto profile = trainerService.findTrainerByUsername(username);
+        if (profile != null) {
```

Comment:
it is nice to have here Optional class, or throw from service custom EntityNotFoundException if repository returns nothing